### PR TITLE
test(db): add test for db

### PR DESCRIPTION
**Pull request by lurais**
**What does this PR do?**
add several test cases to cover more code verification scenarios

**Why are these changes required?**
improve test case coverage

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: framework/src/test/java/org/tron/core/db/AbiStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  private static final SmartContractOuterClass.SmartContract.ABI SOURCE_ABI = jsonStr2Abi(
+      "[{\"inputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\""
+          + ":\"constructor\"}]");
+
+  static {
+    dbPath = "output_AbiStore_test";
+    Args.setParam(
+        new String[]{
+            "--output-directory", dbPath,
+            "--storage-db-directory", dbDirectory
```
Comment (forfreeday) [inline framework/src/test/java/org/tron/core/db/AbiStoreTest.java:40]: What is your intention with the variable `--storage-db-directory`? The storage doesn't do anything.

Comment (lurais) [reply framework/src/test/java/org/tron/core/db/AbiStoreTest.java]: That is not used now, updated

File: framework/src/test/java/org/tron/core/db/AbiStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  @Test
+  public void putTest() {
+    abiStore.put(contractAddr, new AbiCapsule(SOURCE_ABI));
+    Assert.assertEquals(abiStore.has(contractAddr), Boolean.TRUE);
+  }
+
+  @Test
+  public void get() {
+    abiStore.put(contractAddr, new AbiCapsule(SOURCE_ABI));
```
Comment (forfreeday) [inline framework/src/test/java/org/tron/core/db/AbiStoreTest.java:53]: Test method names need to be consistent

File: framework/src/test/java/org/tron/core/db/AbiStoreTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import org.tron.core.config.args.Args;
+import org.tron.core.store.AbiStore;
+import org.tron.core.store.AccountIndexStore;
+import org.tron.protos.Protocol.AccountType;
+import org.tron.protos.contract.SmartContractOuterClass;
+
+public class AbiStoreTest extends BaseTest {
+
+  private static String dbDirectory = "db_AbiStore_test";
+  @Resource
```
Comment (forfreeday) [inline framework/src/test/java/org/tron/core/db/AbiStoreTest.java:25]: `dbDirectory` is not used and should be deleted.

File: framework/src/test/java/org/tron/core/db/AccountTraceStoreTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import org.tron.core.exception.BadItemException;
+import org.tron.core.exception.ItemNotFoundException;
+import org.tron.core.store.AccountIndexStore;
+import org.tron.core.store.AccountTraceStore;
+import org.tron.protos.Protocol.AccountType;
+
+public class AccountTraceStoreTest extends BaseTest {
+
+  private static String dbDirectory = "db_AccountTraceStore_test";
+  @Resource
```
Comment (forfreeday) [inline framework/src/test/java/org/tron/core/db/AccountTraceStoreTest.java:25]: `dbDirectory` is not being used and should be deleted. There are a lot of the same issues behind it, please fix them all together.

File: framework/src/test/java/org/tron/core/db/AccountTraceStoreTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            "--output-directory", dbPath
+        },
+        Constant.TEST_CONF
+    );
+  }
+
+
+  @Test
+  public void recordBalanceWithBlockTest() throws BadItemException, ItemNotFoundException {
+    accountTraceStore.recordBalanceWithBlock(address,1,9999);
```
Comment (forfreeday) [inline framework/src/test/java/org/tron/core/db/AccountTraceStoreTest.java:42]: Keep the same code style

**Review by halibobo1205 — APPROVED** [review]

File: framework/src/test/java/org/tron/core/db/AbiStoreTest.java (hunk 2 above)
Comment (Goodmanisrael27) [reply framework/src/test/java/org/tron/core/db/AbiStoreTest.java:51]: [](https://github.com/polkadot-js/wasm/pull/567)

Comment (Goodmanisrael27) [reply framework/src/test/java/org/tron/core/db/AbiStoreTest.java:51]: > [](https://github.com/polkadot-js/wasm/pull/567)

**Pull request merged**